File: frontend/src/app/models/user-dto.ts
export interface UserDTO {
    idUser: number;
    matricule: string;
    nom: string;
    prenom: string;
    email: string;
    num: string;
    status: UserStatus;
    genre: Genre;
    modifieLe: string; // LocalDateTime est sérialisé en chaîne ISO
    groupeNom: string | null;
}

export enum UserStatus {
    ACTIVE = "ACTIVE",
    INACTIVE = "INACTIVE",
    DELETED = "DELETED"
}

export enum Genre {
    FEMME = "FEMME",
    HOMME = "HOMME"
}

File: backend/src/main/java/com/pfe/backend/DTO/UserDTO.java
package com.pfe.backend.DTO;
import com.pfe.backend.Model.User;
import lombok.*;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class UserDTO {
    private Long idUser;
    private String matricule;
    private String nom;
    private String prenom;
    private String email;
    private String num;
    private User.UserStatus status;
    private User.UserGenre genre;
    private LocalDateTime modifieLe;
    private String groupeNom;
}
